framework/source/class/qx/core/Object.js

```diff
@@ -230,15 +230,30 @@ qx.Class.define("qx.core.Object",
     /**
      * Load user defined data from the object
      *
-     * @param key {String} the key
-     * @return {Object} the user data
+     * @param key {String?} the key
+     *   If key is null or undefined, pass the user the entire map
+     * @param bCopy {Boolean?}
+     *   If bCopy is truthy pass the user a deep clone
+     * @return {Object}
+     *   Returns null if no user data exists or if the given key has
+     *   not been set, the user data map as an object if requested, or
+     *   the value of the user data with the given key
      */
-    getUserData : function(key)
+    getUserData : function(key, bCopy)
     {
-      if (!this.__userData) {
-        return null;
+      var data;
+      if (this.__userData == null) {
```

Comment:
are your shure that you mean this.__userData in this place? IMHO it should be key...

Reply:
Oops good catch! Fixed.